packages/shared/src/utils/route/route.tsx

```diff
@@ -0,0 +1,57 @@
+// Checks if pathname matches route. (Works even with query string /?)
+
+// TODO: Add test cases for this
+import React from 'react';
+import { Redirect } from 'react-router-dom';
+import { routes } from '../routes';
+
+export type TRoute = {
+    component?: React.ElementType | null | ((routes?: TRoute[]) => JSX.Element) | typeof Redirect;
+    default?: boolean;
+    exact?: boolean;
+    getTitle?: () => string;
+    icon_component?: string;
+    id?: string;
+    is_authenticated?: boolean;
+    is_invisible?: boolean;
+    path?: string;
+    to?: string;
+    icon?: string;
+    is_disabled?: boolean;
+    subroutes?: TRoute[];
+    routes?: TRoute[];
+};
```

Comment:
@yauheni-deriv Since all the properties are optional, there is a `Partial<>` can be, right? To your taste
```suggestion
export type TRoute = Partial<{
    component: React.ElementType | null | ((routes?: TRoute[]) => JSX.Element) | typeof Redirect;
    default: boolean;
    exact: boolean;
    getTitle: () => string;
    icon_component: string;
    id: string;
    is_authenticated: boolean;
    is_invisible: boolean;
    path: string;
    to: string;
    icon: string;
    is_disabled: boolean;
    subroutes: TRoute[];
    routes: TRoute[];
}>;
```